### PR TITLE
Don't rely on localStorage when checking for non-Outbrain-compliant tests

### PR DESCRIPTION
**Pull request by joelochlann**
## Bug
We were sometimes flagging a page as Outbrain-compliant (meaning nothing intervenes between the bottom of the article and Outbrain), when in fact the epic was being displayed so the article was not compliant.

This problem is intermittent. It is seems to be a race condition between the code which checks for Outbrain compliance and the code which sets A/B test participations in localStorage. When `isUserInContributionsAbTest()` runs, the epic A/B test participation is sometimes not yet in localStorage even though the test will run on that page.

I'm still not sure what changed on the site recently to bring this issue to light, but given that it's a timing-sensitive issue I imagine it could have been caused by many seemingly unrelated changes. For instance, changing dependencies between modules can cause webpack to put a given module in a different bundle. It might be that some relevant code moved from a bundle which was guaranteed to load later into a bundle which loads in parallel with the Outbrain code.

## Fix
We now use `getAcquisitionTest()` to in the Outbrain compliance check. This ultimately uses `variantFor` and `isInTest` from `segment-util.js`, which computes test participations directly from the MVT cookie rather than localStorage.

This actually has another benefit. As far as I can tell, previously we never actually respected the epic view conditions in this check (e.g. that the default "Ask Four" epic won't display on more than four pageviews in a month). So we were probably overzealous with flagging non-compliance on pages where the epic wouldn't display.

I've tested on CODE (I couldn't reproduce the original issue locally, probably because the bundles are assembled differently in dev mode) and having made this change, the compliance check gave the right result every time I checked (about 20 times), whereas before it erroneously flagged the page as compliant at least half the time. So it's at least an improvement.

### CODE REVIEW

**Review by svillafe — APPROVED** [review]
Looks much simpler than before!! Good work @joelochlann !

File: static/src/javascripts/projects/commercial/modules/check-dispatcher.js [hunk 1]
```diff
@@ -2,15 +2,11 @@
...
-import {
-    emailTests,
-    contributionsTests,
-    userIsInAClashingAbTest,
-} from 'common/modules/experiments/ab-test-clash';
 import { trackAdRender } from 'commercial/modules/dfp/track-ad-render';
 import { commercialFeatures } from 'common/modules/commercial/commercial-features';
 import { checks } from 'common/modules/check-mediator-checks';
 import { resolveCheck, waitForCheck } from 'common/modules/check-mediator';
+import { getTest as getAcquisitionTest } from 'common/modules/experiments/acquisition-test-selector';
```
Comment (jranks123) [inline static/src/javascripts/projects/commercial/modules/check-dispatcher.js]: Out of interest why do we have this file and the identical `static/src/javascripts/projects/commercial-control/modules/check-dispatcher.js` file?

Comment (rich-nguyen) [reply static/src/javascripts/projects/commercial/modules/check-dispatcher.js]: `commercial-control` is run on a 1% control group to measure the difference between today's commercial system and the "start of the quarter code". See this PR for more info: https://github.com/guardian/frontend/pull/18677 . And here's how we are doing today (not good, due to Garnett): https://frontend.gutools.co.uk/commercial/kpi/revenue

**Comment by jranks123** [comment]
@joelochlann Once you have seen this epic 4 times, your `gu.ab.participants` in Local Storage no longer contains the test, so I don't think this would have been the reason why we were flagging non-compliance when the view-check failed.

File: static/src/javascripts/projects/commercial-control/modules/check-dispatcher.js [hunk 2]
```diff
@@ -2,15 +2,11 @@
...
-import {
-    emailTests,
-    contributionsTests,
-    userIsInAClashingAbTest,
-} from 'common/modules/experiments/ab-test-clash';
 import { trackAdRender } from 'commercial-control/modules/dfp/track-ad-render';
 import { commercialFeatures } from 'common/modules/commercial/commercial-features';
 import { checks } from 'common/modules/check-mediator-checks';
 import { resolveCheck, waitForCheck } from 'common/modules/check-mediator';
+import { getTest as getAcquisitionTest } from 'common/modules/experiments/acquisition-test-selector';
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/commercial-control/modules/check-dispatcher.js]: thanks for fixing up `commercial-control` too. It's used by a 1% group of the audience, so it is worth copying your fixes too. It's kind of like back-porting your fix. Ordinarily, by fixing just files in `commercial` folder, you are fixing behaviour for 99% of the audience.

File: static/src/javascripts/projects/common/modules/experiments/ab-test-clash.js [hunk 3]
```diff
@@ -1,32 +0,0 @@
-// @flow
-import { isInVariant } from 'common/modules/experiments/utils';
-import { abTestClashData as acquisitionsAbTestClashData } from 'common/modules/experiments/acquisition-test-selector';
-
```
Comment (jranks123) [inline static/src/javascripts/projects/common/modules/experiments/ab-test-clash.js]: So happy to see this file die

File: static/src/javascripts/projects/commercial-control/modules/check-dispatcher.js (hunk 2 above)
Comment (rich-nguyen) [reply static/src/javascripts/projects/commercial-control/modules/check-dispatcher.js]: definite overuse of the word fix.

**Review by rich-nguyen — APPROVED** [review]

**Review by jranks123 — APPROVED** [review]

**Comment by joelochlann** [comment]
@jranks123 that's a good point. Probably the same race condition would have applied on the pageview in which the test should be removed from localStorage, but given that the number of people who reach that fourth pageview with a given browser is much smaller than those with a first pageview, it wouldn't have been enough to balance out

**Pull request merged**